Release/include/cpprest/details/basic_types.h

```diff
@@ -43,20 +48,358 @@ typedef uint64_t size64_t;
 typedef uint32_t HRESULT; // Needed for PPLX
 #endif
 
+#if defined _WIN32 && !defined _DLL
+// TEMPLATE CLASS allocator
+template<class _Ty>
+class allocator : public std::allocator<_Ty> {
+public:
+    typedef allocator<_Ty> _Alloc;
+    typedef std::allocator<_Ty> _Mybase;
+
+    template<class _Other>
+    struct rebind {
+        // convert this type to allocator<_Other>
+        typedef allocator<_Other> other;
+    };
+
+    allocator() CPPREST_NOEXCEPT : _Mybase() {
+        // construct default allocator (do nothing)
+    }
+
+    allocator(const _Alloc& _Right) CPPREST_NOEXCEPT : _Mybase(_Right) {
+        // construct by copying (do nothing)
+    }
+
+    allocator(_Alloc&& _Right) CPPREST_NOEXCEPT : _Mybase(std::move(_Right)) {
+        // construct by moving (do nothing)
+    }
+
+    template<class _Other>
+    allocator(const allocator<_Other>& _Right) CPPREST_NOEXCEPT : _Mybase(_Right) {
+        // construct from a related allocator (do nothing)
+    }
+
+    template<class _Other>
+    allocator(allocator<_Other>&& _Right) CPPREST_NOEXCEPT : _Mybase(std::forward<_Other>(_Right)) {
+        // construct from a related allocator (do nothing)
+    }
+
+    _Alloc& operator=(const _Alloc& _Right) {
+        // assign by copying
+        _Mybase::operator=(_Right);
+        return (*this);
+    }
+
+    _Alloc& operator=(_Alloc&& _Right) {
+        // assign by moving
+        _Mybase::operator=(std::move(_Right));
+        return (*this);
+    }
+
+    template<class _Other>
+    _Alloc& operator=(const allocator<_Other>&) {
+        // assign from a related allocator
+        _Mybase::operator=(std::forward<_Other>(_Right));
+        return (*this);
+    }
+
+    _DECLSPEC_ALLOCATOR pointer allocate(size_type _Count) {
+        // allocate array of _Count elements
+        return static_cast<pointer>(::HeapAlloc(::GetProcessHeap(), NULL, _Count * sizeof(_Ty)));
```

Comment:
This allocate is damaged vs. std::allocator -- it is not checking for overflow of _Count.

Reply:
Would the following resolve this ?
```
__declspec(allocator) allocate(size_type _Count) {
        // allocate array of _Count elements
	if (_Count == 0)
		return nullptr;

	// check overflow of multiply
	if ((size_t)(-1) / sizeof(_Ty) < _Count)
		_Xbad_alloc();	// report no memory

	return static_cast<pointer>(::HeapAlloc(::GetProcessHeap(), NULL, _Count * sizeof(_Ty)));
}
```